internal/querycoord/query_coord.go

```diff
@@ -514,7 +516,27 @@ func (qc *QueryCoord) watchHandoffSegmentLoop() {
 		select {
 		case <-ctx.Done():
 			return
-		case resp := <-watchChan:
+		case resp, ok := <-watchChan:
+			if !ok {
+				log.Warn("QueryCoord watch handoff segment loop failed because watch channel is closed")
+				return
```

Comment:
Shall we panic here? Since if this goroutine quit, no one will watch hand off segment events

Reply:
done